packages/utils/src/lib/log-results.ts

```diff
@@ -0,0 +1,53 @@
+type PromiseFulfilledResult<T> = {
+  status: 'fulfilled';
+  value: T;
+};
+
+export function logMultipleResults<T>(
+  results: PromiseSettledResult<T>[],
+  messagePrefix: string,
+  succeededCallback?: (result: PromiseFulfilledResult<T>) => void,
+  failedCallback?: (result: PromiseRejectedResult) => void,
+) {
+  if (succeededCallback) {
+    const succeededResults = results.filter(
+      (result): result is PromiseFulfilledResult<T> =>
+        result.status === 'fulfilled',
+    );
+
+    logPluginExecution(
+      succeededResults,
+      `${messagePrefix} successfully: `,
+      succeededCallback,
+    );
+  }
+
+  if (failedCallback) {
+    const failedResults = results.filter(
+      (result): result is PromiseRejectedResult => result.status === 'rejected',
+    );
```

Comment:
Same as for fulfilled results, a type guard in `utils` could be useful:

```ts
export function isPromiseRejectedResult(
  result: PromiseSettledResult<unknown>,
): result is PromiseRejectedResult {
  return result.status === 'rejected';
}
```

```suggestion
    const failedResults = results.filter(isPromiseRejectedResult);
```

Reply:
Added guard